qa/tasks/cephfs/fuse_mount.py

```diff
@@ -439,12 +439,21 @@ def find_socket(client_name):
             timeout=(15*60)).strip()
         log.info("Found client admin socket at {0}".format(asok_path))
 
-        # Query client ID from admin socket
-        json_data = self.client_remote.sh(
-            ['sudo', self._prefix + 'ceph', '--admin-daemon', asok_path] + args,
-            stdout=StringIO(),
-            timeout=(15*60))
-        return json.loads(json_data)
+        # Query client ID from admin socket, wait and
+        # retry 5 times if it is not ready
+        for i in range(10):
+            try:
+                p = self.client_remote.run(args=
+                    ['sudo', self._prefix + 'ceph', '--admin-daemon', asok_path] + args,
+                    stdout=StringIO(), stderr=StringIO(),
+                    timeout=(15*60))
+                break
+            except CommandFailedError:
+                if "Connection refused" in stderr.getvalue():
```

Comment:
Rather than running looping in case of `Connection refused` perhaps we should add `sleep(10)` before running `admin_socket()` in `gather_mount_info()` since you hit this error when vstart_runner.py calls `gather_mount_info()` for the very first time.

Otherwise every command that fails with `Connection refused` error message would have to wait for 20 seconds.

Reply:
10 senconds are two long, especially for the test case need to do mount/umount very frequently.

BTW, is there any reason that the `Connection refused` is expected ? In this loop it will just try 10 times and each will sleep 2 seconds, locally from my test it will just retry twice at most, that mean it will only wait 2 seconds at most.

This will also be potentially bug for none vstart_runner.py case in theory.

Reply:
> 10 senconds are two long, especially for the test case need to do mount/umount very frequently.

We could wait for lesser amount of time. I didn't mean 10 seconds necessarily. I suspect from the screenshot I saw in G-Chat that this might be cause Ceph cluster wasn't fully up.

Reply:
From my local test mostly the first time will be successful, and randomly we need twice. Yesterday I have also seen one time it retried 6 times, that means we need to wait at least 12 seconds or more ?

So the loop is needed IMO, because we couldn't know how long to wait will be proper. Use the loop we can be sure that for most cases, we won't wait for a fixed 5 seconds or so.

Reply:
Ok, I see. Since the time required varies, there's no other way than using a loop. Cool. Marking the conversation resolved.